storage/storage.go

```diff
@@ -113,13 +113,47 @@ var (
 			return 1
 		`)
 
+	// Atomically get an aggregation
+	//
+	// This should only return an aggregation if it is available for processing.
+	// This is determined by the ExpiresAt key which contains a Unix Timestamp.
+	// ExpiresAt:
+	//   - false                        : Aggregation not found
+	//   - ""						    : First time seeing this aggregation
+	// 								      set ExpiresAt based on timeout and return it
+	//   - ExpiresAt + timeout < now    : Aggregation has expired and thus is available for processing
+	//                                    set ExpiresAt based on timeout and return it
+	//   - ExpiresAt + timeout >= now   : Aggregation is already being processed. Return error "BEINGPROCESSED"
```

Comment:
When we set the **ExpiresAt** value we add the timeout to the current timestamp so couldn't we simplify those checks to something like:

```
ExpiresAt < now  :: aggregation timeout expired
ExpiresAt >= now :: aggregation is still being processed 
```

Is there something I'm missing in the current logic?